extensions-contrib/prometheus-emitter/src/main/java/org/apache/druid/emitter/prometheus/PrometheusEmitter.java

```diff
@@ -87,7 +95,7 @@ public void start()
         log.error("HTTPServer is already started");
       }
       // Start TTL scheduler if TTL is configured
-      if (config.getFlushPeriod() != null) {
+      if (config.getFlushPeriod() != null && exec == null) {
         exec = ScheduledExecutors.fixed(1, "PrometheusTTLExecutor-%s");
```

Comment:
Not a blocker since this is existing code... we could perhaps move the initialization of `exec` in the constructor and make it a `final` variable. `start()` would then just do the execution start